client/src/components/Map/components/SuperBlock.js

```diff
@@ -90,7 +94,49 @@ export class SuperBlock extends Component {
   }
 
   render() {
-    const { superBlock, isExpanded, toggleSuperBlock } = this.props;
+    const {
+      completedChallenges,
+      superBlock,
+      nodes,
+      isExpanded,
+      toggleSuperBlock
+    } = this.props;
+
+    // since the nodes have been filtered based on isHidden, any blocks whose
+    // nodes have been entirely removed will not appear in this array.
+    const blocksForSuperBlock = nodes.filter(
+      node => node.superBlock === superBlock
+    );
+
+    const certificationProjects = blocksForSuperBlock.filter(block =>
+      projectDashedBlockNameRE.test(block.block)
+    );
+
+    const completedProjectCount = certificationProjects.filter(block =>
+      completedChallenges.some(completedId => block.id === completedId)
+    ).length;
+
+    const completedLessonCount =
+      blocksForSuperBlock.filter(block =>
+        completedChallenges.some(completedId => block.id === completedId)
+      ).length - completedProjectCount;
+
+    // Simply using the % of completed projects will work, but since the only
+    // requirement for getting a certificate is to complete all the projects,
+    // the progress tracker should reflect this.
+    // To advance the tracker normally for users who only complete the projectsa
+    // fill the progress by
+    // (% of lessons completed) + (empty space) * (% of projects completed)
+    //
+    // As an equation:
+    // l = (completed lessons)/(number of lessons + projects)
+    // p = (completed projects)/(number of projects)
+    // progress = l + (1 - l)*p
+    //          = l + p - l*p
+    const l = completedLessonCount / blocksForSuperBlock.length;
+    const p = completedProjectCount / certificationProjects.length;
+    const progress = l + p - l * p;
```

Comment:
The check mark is displayed only if:
- `l = 100%` (All lessons + projects are complete) OR
- `p = 100% (All projects are complete)`

Please let me know if I should add a more detailed explaination.

Reply:
The explanations are very clear, don't worry about that!  It's just that I think there should be visual difference between `l == 1` and `p == 1`.  We don't want people who complete all the challenges, but don't do any of the projects, to think that there is nothing left for them to do.

Perhaps the least confusing thing would be to use one of the other ideas you mentioned on Gitter, namely adjust the formula to this:

```
progress = k * l + p * (1 - k * l)
```

where `0 < k < 1` (probably 0.75).  I think that's better than what I suggested, since this way it's completely obvious that the completion of all the challenges is not the end of a course.

Reply:
> The explanations are very clear

I don't think they are.
Let me try this again:
`l == 1` only if ALL then challenges (lessons + projects) are complete and a check is displayed.
`p == 1` only if all the PROJECTS are complete and a check is displayed.
In all other cases, the check is NOT displayed.

Reply:
Ah, my mistake - assumed I understood and skimmed the instructions.  Sorry about that!  Let's do a quick example to make sure we're on the same page.  

Assume there are 95 challenges and 5 projects.  If the user does all of the challenges, they end up with a `progress` of 0.95.  If the user does 1 project, they end up with `progress` of 0.208 (because the project appears in both `l` and `p`).

What I would prefer is the following.  In the first case they should have a progress of k (0.75, say) and the second they should have exactly 0.2 (i.e. projects should not be double-counted).  We can achieve this by doing this:

```js
    const lessonCount = blocksForSuperBlock.length;
    const projectCount =  certificationProjects.length;
    const challengeCount = lessonCount - projectCount;
    const c = completedChallengeCount / challengeCount;
    const p = completedProjectCount / projectCount;
    const k = 0.75;
    const progress = k * c + p * (1 - k * c);
```
where I've shifted to using `c` to avoid confusing challenges with 'lessons'.

Reply:
> If the user does 1 project, they end up with progress of 0.208 (because the project appears in both l and p).

No, the progress increases by `0.2` exactly, because projects are NOT double counted.

```javascript
    const completedLessonCount =
      blocksForSuperBlock.filter(({ id }) => completedChallenges.includes(id))
        .length - completedProjectCount;
```
This would make the progress bar (circle?) fill smoothly and equally for every step, both for users who do only projects and for users who do both lessons and projects.
I feel that the third possibility of users doing some projects and some lessons is quite rare, but this would work for them as well.
It is still obvious that completing the lessons is not completing the course, since there is still a sliver incomplete.
I feel that this is more encouraging, since the user can see that they are almost done :)

Now, if you want to fill the progress bar more for projects and less for lessons then that is different.

Reply:
Both equations would work well, the only difference is the where they end up. I don't think it's a deal breaker, so let's leave both options on the table and see what other reviewers prefer.